tests/adapters/test_http.py

```diff
@@ -0,0 +1,103 @@
+import pytest
+from aiohttp import web
+from mock import Mock, PropertyMock, create_autospec
+
+from tickit.adapters.httpadapter import HTTPAdapter
+from tickit.adapters.servers.http_server import HTTPServer
+from tickit.core.device import Device
+
+
+@pytest.fixture
+def MockDevice() -> Mock:
+    return create_autospec(Device, instance=False)
+
+
+@pytest.fixture
+def mock_device() -> Mock:
+    return create_autospec(Device, instance=True)
+
+
+@pytest.fixture
+def MockServer() -> Mock:
+    return create_autospec(HTTPServer, instance=False)
+
+
+@pytest.fixture
+def mock_server() -> Mock:
+    return create_autospec(HTTPServer, instance=True)
+
+
+@pytest.fixture
+def MockServerConfig() -> Mock:
+    return create_autospec(HTTPServer.HTTPServerConfig, instance=False)
+
+
+@pytest.fixture
+def mock_server_config() -> Mock:
+    return create_autospec(HTTPServer.HTTPServerConfig, instance=True)
```

Comment:
Return type should be the mocked type rather than `Mock`, for example:

```python
def MockDevice() -> Device:
```

Reply:
Garry did this for test_composed.py.... :eyes:

Reply:
It works, obviously, but mypy/vscode are happier with the types you expect to be working with

Reply:
There we go, another thing for you to fix :stuck_out_tongue: